Assets/Script/Gameplay/HUD/Practice/PracticeSectionView.cs

```diff
@@ -101,5 +105,223 @@ public void UpdateView()
                 _background.color = Color.clear;
             }
         }
+
+        private string ParseSectionName(string sectionName)
+        {
+            const string SECTION_PREFIX = "section ";
+            const string PRC_PREFIX = "prc_";
+
+            if (sectionName.StartsWith(SECTION_PREFIX))
+            {
+                return sectionName[SECTION_PREFIX.Length..].TrimStart('_').Trim();
+            }
+
+            if (sectionName.StartsWith(PRC_PREFIX))
+            {
+                return ParsePrcSectionName(sectionName[PRC_PREFIX.Length..]);
+            }
+
+            return sectionName;
+        }
+
+        // Receives a prc-based section name with "prc_" removed
+        private string ParsePrcSectionName(string prcSectionName)
+        {
+            // Handle letter-based sections like "A section" (prc_a) and "B section 3" (prc_b3)
+            if (isLetterBasedSectionName(prcSectionName))
+            {
+                var name = char.ToUpper(prcSectionName[0]) + " " + Localize.Key("Gameplay.PracticeSections.Section");
+                return prcSectionName.Length == 1 ? name : (name + " " + prcSectionName[1..]);
+            }
+
+            foreach (var key in prcNameToLocalizationKey.Keys)
+            {
+                if (key == prcSectionName)
+                {
+                    return Localize.Key("Gameplay.PracticeSections." + prcNameToLocalizationKey[key]);
+                }
+
+                if (prcSectionName.StartsWith(key))
+                {
+                    var remainder = prcSectionName[(key.Length + "_".Length)..];
+
+                    if (isSectionNumber(remainder))
+                    {
+                        return Localize.Key("Gameplay.PracticeSections." + prcNameToLocalizationKey[key]) + " " + remainder;
```

Comment:
Again here, the localization string should be a format string instead of manually constructed.

```suggestion
                        return Localize.KeyFormat(("Gameplay.PracticeSections", prcNameToLocalizationKey[key], "WithNumber"), remainder);
```

This would somewhat complicate the localization strings for each section, as both no-number and with-number variants would need to be provided for each.

Reply:
I suppose we'd want to move `LetterSection` and `WithNumber` out of the same grouping as the actual section flags, to avoid false positives if someone actually uses something like `[section WithNumber]` for some reason. `Gameplay.Practice.Sections` for the literal flags and `Gameplay.Practice.SectionFormats` for those two?

Reply:
Pushed a commit with this approach implemented - let me know what you think 🙂